src/AspNetCore/Src/inprocessapplication.cxx

```diff
@@ -684,3 +698,83 @@ IN_PROCESS_APPLICATION::FindHighestDotNetVersion(
     // we check FAILED(hr) outside of function
     return hr;
 }
+
+//
+// Runs a standalone appliction.
+// The folder structure looks like this:
+// Application/
+//   hostfxr.dll
+//   Application.exe
+//   Application.dll
+//   etc.
+// We get the full path to hostfxr.dll and Application.dll and run hostfxr_main,
+// passing in Application.dll.
+// Assuming we don't need Application.exe as the dll is the actual application.
+//
+HRESULT
+IN_PROCESS_APPLICATION::RunStandaloneApplication()
+{
+    HRESULT     hr = S_OK;
+
+    STRU                        struApplicationDllPath;
+    STRU                        struHostfxrPath;
+    PCWSTR                      pszDotnetExeLocation = NULL;
+    DWORD                       dwPosition;
+    HMODULE                     hModule;
+    PCWSTR                      argv[2];
+    hostfxr_main_fn             pProc;
+
+    PATH::ConvertPathToFullPath(m_pConfiguration->QueryProcessPath()->QueryStr(),
+        m_pConfiguration->QueryApplicationFullPath()->QueryStr(),
+        &struApplicationDllPath);
+
+    // Remove Application.dll and append hostfxr.dll
+    dwPosition = struApplicationDllPath.LastIndexOf(L'\\', 0);
+    hr = struHostfxrPath.Copy(struApplicationDllPath);
+    if (FAILED(hr))
+    {
+        goto Finished;
+    }
+
+    struHostfxrPath.QueryStr()[dwPosition] = L'\0';
+    struApplicationDllPath.QueryStr()[dwPosition] = L'\0';
+
+    // Whenever we directly modify the stru buffer, we need to call SyncWithBuffer
+    if (FAILED(hr = struHostfxrPath.SyncWithBuffer())
+        || FAILED(hr = struHostfxrPath.Append(L"\\hostfxr.dll"))
+        || FAILED(hr = struApplicationDllPath.SyncWithBuffer())
+        || FAILED(hr = struApplicationDllPath.Append(L"\\NativeIISSample.dll")))
```

Comment:
?

Reply:
Oops, forgot to actually parse that out XD